s4e-web/src/app/utils/initializer/config.service.spec.ts

```diff
@@ -1,44 +1,49 @@
 import {TestBed} from '@angular/core/testing';
 import {S4eConfig} from './config.service';
 import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
-import {IConfiguration} from '../../app.configuration';
+import {SessionService} from '../../state/session/session.service';
+import {SessionFactory} from '../../state/session/session.factory.spec';
+import {of} from 'rxjs';
 
 describe('Configuration service', () => {
   let configService: S4eConfig;
   let http: HttpTestingController;
+  let sessionService: SessionService;
 
   beforeEach(() => {
     TestBed.configureTestingModule({
-      providers: [S4eConfig],
+      providers: [S4eConfig, SessionService],
       imports: [HttpClientTestingModule]
     });
 
     configService = TestBed.get(S4eConfig);
+    sessionService = TestBed.get(SessionService);
     http = TestBed.get(HttpTestingController);
   });
 
   it('should be created', () => {
     expect(configService).toBeDefined();
   });
 
-  it('should call endpoint', (done) => {
-    configService.loadConfiguration().then(() => {
-      expect(configService.geoserverUrl).toBe('localhost:8080/test/wms');
-      expect(configService.geoserverWorkspace).toBe('testWorkspace');
-      expect(configService.recaptchaSiteKey).toBe('key');
-      done();
-    });
-
-    const req = http.expectOne('api/v1/config');
+  it('should call endpoint', async () => {
+    const spy = spyOn(sessionService, 'getProfile$').and.returnValue(of(SessionFactory.build()));
+    const p = configService.loadConfiguration();
 
-    req.flush({
+    const reqConfig = http.expectOne('api/v1/config');
```

Comment:
Why you're not using `s4eConfig` service to get `API_PREFIX`?

Reply:
That's one place where it is hardcoded, as we're testing it here. At this point It is not initialized.